### PR TITLE
Adding key words for better google results for index.mdx

### PR DESCRIPTION
**Pull request by ramramhariram**
currently these are hidden in our Consul enterprise page which is not necessarily what everyone is searching for, i see customers/users searching for things like 'compatibiity matrix' or 'feature matrix' to pull this kind of info for consul and I think adding in 'matrix' will help with google searches.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] not a security concern

### CODE REVIEW

**Review by im2nguyen — APPROVED** [review]
thank you!

**Comment by jkirschner-hashicorp** [comment]
@boruszak : If we want to improve search results that include "matrix" in the query, what's the right way to accomplish that? Would it be altering the page metadata rather than specific header titles?

Note: If we did change the header titles, we should consider adding anchor aliases to we don't break existing links to these sections.

**Comment by im2nguyen** [comment]
Good call, we could definitely add "Evaluate Enteprise features with the feature availability and compatibility matrix." to the metadata (description) so it appears in the the devdot search. Updating the header only will let folks do a "control + f" on the page.

Google doesn't treat headers differently for SEO [source](https://www.searchenginejournal.com/heading-tags-for-seo/341817/)... but if "compatibility matrix" is what folks are searching for, if it appears on the page, it would rank higher.

File: website/content/docs/enterprise/index.mdx [hunk 1]
```diff
@@ -88,7 +88,7 @@ Available Enterprise features per Consul form and license include:
 [HashiCorp Cloud Platform (HCP) Consul]: https://cloud.hashicorp.com/products/consul
 [Consul Enterprise]: https://www.hashicorp.com/products/consul/
 
-### Feature availability by runtime
+### Feature availability matrix by runtime
```
Comment (im2nguyen) [inline website/content/docs/enterprise/index.mdx:91]:
```suggestion
### Feature availability matrix by runtime ((#feature-availability-by-runtime))
```

**Comment by ramramhariram** [comment]
hey @boruszak -  my initial comments are about the specific content that i proposed changes for. The feature matrix that i updated specifically addresses questions like  - "Do Consul EKS servers support clients on ECS"? That is not an enterprise feature question, in fact it is not even about any specific feature, but rather a question about compatibility. How do we drive answers to those kind of questions to this page? Currently any search around 'compatibility' etc goes to our k8s version compatibility or envoy/consul tables because those pages have those keywords in them.

**Comment by ramramhariram** [comment]
yes, and maybe the best option is to have one page called compatibility matrix that houses ALL of compatibility matrices together.

**Pull request closed without merging**